src/audio/copier/ipcgtw_copier.h

```diff
@@ -95,13 +95,22 @@ struct ipc4_ipc_gateway_cmd_data_reply {
 int copier_ipcgtw_process(const struct ipc4_ipcgtw_cmd *cmd,
 			  void *reply_payload, uint32_t *reply_payload_size);
 
-int copier_ipcgtw_params(struct ipcgtw_data *ipcgtw_data, struct comp_dev *dev,
-			 struct sof_ipc_stream_params *params);
-
-void copier_ipcgtw_reset(struct comp_dev *dev);
-
 int copier_ipcgtw_create(struct comp_dev *dev, struct copier_data *cd,
 			 const struct ipc4_copier_module_cfg *copier, struct pipeline *pipeline);
 
+#if CONFIG_IPC4_GATEWAY
 void copier_ipcgtw_free(struct copier_data *cd);
+int copier_ipcgtw_params(struct ipcgtw_data *ipcgtw_data, struct comp_dev *dev,
```

Comment:
Shouldn't this be fixed in copier.c, or if done here, ifdef out all copier_ipcgtw_foo calls? This feels a bit adhoc to ifdef some functions out. Might be easier to verify in copier.c that code still works correctly if the feature is disabled (or gives proper errors if we reach branches of code that we really shouldn't when gateway support is not built in).